src/planner/binder/tableref/bind_basetableref.cpp

```diff
@@ -244,8 +233,25 @@ unique_ptr<BoundTableRef> Binder::Bind(BaseTableRef &ref) {
 		// could not find an alternative: bind again to get the error
 		// note: this will always throw when using DuckDB as a catalog, but a second look-up might succeed
 		// in catalogs that do not have transactional DDL
-		table_or_view =
-		    entry_retriever.GetEntry(ref.catalog_name, ref.schema_name, table_lookup, OnEntryNotFound::THROW_EXCEPTION);
+		try {
+			table_or_view = entry_retriever.GetEntry(ref.catalog_name, ref.schema_name, table_lookup,
```

Comment:
Running this just to catch the throw seems unnecessary - can we just throw the exception prior to running this?